From 16294c6d5b364ef9b46f204b8d2df0f673b845d9 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 20 Mar 2019 08:56:40 +0200
Subject: [PATCH] [SW-0] spi: use per device chip select

Read per device CS not from a common place since it located in a
different IO mapping

NOTE: mach/hl_regs.h does not provide mmPSOC_GLOBAL_CONF_PAD_SEL_53
anymore
Need to include the specific device header
Change-Id: I2428d5dc773476a2a21a404fe6eede8ea1315460
---

diff --git a/common/spi.c b/common/spi.c
index 1315bd2..4deb52a 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -11,6 +11,7 @@
 #include <linux/bitops.h>
 #include <mach/hl_regs.h>
 
+#include "spi.h"
 #include "uart.h"
 
 #define mmPSOC_SPI_CTRLR0		(HL_CFG_BASE + 0xC43000)
@@ -63,12 +64,22 @@
 
 static void spi_claim_bus(void)
 {
-	writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_53);
+	static u32 *reg;
+
+	if (!reg)
+		reg = get_cs_pad_sel();
+
+	writel(1, reg);
 }
 
 static void spi_release_bus(void)
 {
-	writel(0, mmPSOC_GLOBAL_CONF_PAD_SEL_53);
+	static u32 *reg;
+
+	if (!reg)
+		reg = get_cs_pad_sel();
+
+	writel(0, reg);
 }
 
 static inline void spi_dev_update_frame_size(unsigned int len)
diff --git a/common/spi.h b/common/spi.h
index 5d2051c..a465f1e 100644
--- a/common/spi.h
+++ b/common/spi.h
@@ -10,5 +10,6 @@
 
 
 void spi_copy_spl(unsigned int spl_offset);
+unsigned int *get_cs_pad_sel(void);
 
 #endif /* _HL_SPI_H_ */
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index de6092a..c05e109 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -7,6 +7,7 @@
 
 #include <linux/io.h>
 #include <mach/hl_regs.h>
+#include <mach/../../gaudi/include/mach/gaudi_regs.h>
 
 void preboot_pcie_wait_for_link(void)
 {
@@ -19,3 +20,9 @@
 			break;
 	}
 }
+
+unsigned int *get_cs_pad_sel(void)
+{
+	return (unsigned int *)mmPSOC_GLOBAL_CONF_PAD_SEL_53;
+}
+
diff --git a/goya/goya.c b/goya/goya.c
index 7ea9ce2..d5e5fe6 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -8,6 +8,7 @@
 #include <common.h>
 #include <linux/io.h>
 #include <mach/hl_regs.h>
+#include <mach/../../goya/include/mach/goya_regs.h>
 
 #define BTL_COUNTER_MAX		BIT(22)
 #define SYS_REF_CLOCK		50000000
@@ -148,3 +149,8 @@
 	writel(1, mmPCIE_DB_RSV_RSV_1);
 }
 
+unsigned int *get_cs_pad_sel(void)
+{
+	return (unsigned int *)mmPSOC_GLOBAL_CONF_PAD_SEL_53;
+}
+
